### DISCUSSION TITLE
Feature Suggestion: Socket Integration and JOT

### DISCUSSION THREAD

**Discussion by ghost** [TodePond/GulfOfMexico#270] [General]
With Dreamberd, sending socket requests is now easier than ever. No quotation marks or any other nonsense needed! Simply add a @ symbol in your code as such
`
const var request@google.com! // The HTML for Google home page
`
This also applies to strings, making it easier to create applications that can interface with the web.
`
const const const email = "hello@example.com"! // The HTML for example.com
`

Additionally, Dreamberd should implement a new JIT feature, the Just Out of Time compiler (JOT.) JOT will compile your DreamBerd code into the latest Scratch version just out of time such that a non-JOT version would have the same performance. Additionally, JOT will support compiling back into DreamBerd, creating an infinite cycle of compilation.

**Comment by ghost** [comment]
You will also be able to create variables with negative lifetimes. This would cause the variable to not be defined, but you will be able to use the `previous` keyword to retrieve it back from the past.

**Reply by stohrendorf** [reply]
See also discussions in #132.

**Comment by ghost** [comment]
![bar-graph](https://github.com/TodePond/DreamBerd/assets/110881816/4ea393a0-0ab6-46bf-9c96-b6a5814f1327)
Just Out of Time compiler

**Comment by ghost** [comment]
Expedited Garbage Collector:
![image](https://github.com/TodePond/DreamBerd/assets/110881816/f5a90cc9-0e62-4f5c-9e84-61cd9b969300)
DreamBerd's GC will nicely collect all of your program's garbage into one neat file which you can clean up.

(This is for NodeBerd, Web Berd will send your trash to wm@wastemanagement.com)
